pkgs/applications/gis/qgis/wrapped.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, lib, fetchurl, openssl, python3Packages, makeWrapper, symlinkJoin
+, qgis3-unwrapped
+}:
+with lib;
+symlinkJoin rec {
+  inherit (qgis3-unwrapped) version;
+  name = "qgis-${version}";
+
+  paths = [ qgis3-unwrapped ];
+
+  nativeBuildInputs = [ makeWrapper python3Packages.wrapPython ];
+
+  # extend to add to the python environment of QGIS without rebuilding QGIS application.
+  pythonInputs = qgis3-unwrapped.pythonBuildInputs;
+
+  # use the source archive directly to avoid rebuilding when changing qgis distro
+  inherit (qgis3-unwrapped) src;
+
+  postBuild = ''
+    unpackPhase
+
+    buildPythonPath "$pythonInputs"
+
+    wrapProgram $out/bin/qgis \
+      --prefix PATH : $program_PATH \
+      --set PYTHONPATH $program_PYTHONPATH
+
+    # desktop link
```

Comment:
The following seems irrelevant for the wrapper and thus better placed in the original derivation. Or am I missing something?